filesystem_service/tests/utils/monitor.py

```diff
@@ -0,0 +1,90 @@
+"""
+Provides useful functions for filesystem monitor tests
+"""
+
+import collections
+from multiprocessing import Manager, Process
+import os
+from time import sleep
+
+from filesystem_service.monitor import Event, FileSystemMonitor
+
+from .consts import SUBDIR_FPATH, SUBDIR_PATH
+
+
+InotifyEvent = collections.namedtuple('_INOTIFY_EVENT',
+                                      ['wd', 'mask', 'cookie', 'len'])
+EVENTS = Manager().list()
+
+
+# pylint: disable=too-few-public-methods
+class Subscriber(object):
+    """
+    Subscriber emulator
+    """
+
+    @staticmethod
+    def update(event):
+        """
+        :param event: inotify event
+        add events to shared list variable
+        """
+
+        EVENTS.append(event)
+
+
+def get_monitor_instance(folder='/test', subscriber=None):
+    """
+    :param folder: folder to be watched
+    :param subscriber: an instance of subscriber
+    :return: an instance of FileSystemMonitor
+    """
+
+    monitor = FileSystemMonitor(folder)
+    if subscriber:
+        monitor.subscribe(subscriber)
+
+    return monitor
+
+
+def gen_event_object():
+    """
+    :return: an instance of Event
+    """
+
+    i_event = (InotifyEvent(wd=1, mask=8, cookie=0, len=16),
+               ['IN_CLOSE_WRITE', 'IN_ACCESS'], b'/tmp/onedrive',
+               b'test')
+
+    return Event(i_event)
+
+
+def start_monitor_process(monitor):
+    """
+    :param monitor: an instance of monitor
+    Starts monitor in its own process in the current thread
+    """
+
+    process = Process(target=monitor.monitor)
```

Comment:
Why not threads?

Reply:
Why not process?

Reply:
Because threads are more easy to synchronize. For example instead of `EVENTS = Manager().list()` you wold just be able to use a regular `list`.

Reply:
Don't see anything complecated here.

Reply:
People who would read it will.